### PR TITLE
Add Filecoin

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

- added: Add Filecoin

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204526951399191

### CODE REVIEW

File: src/filecoin/Filscan.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      headers: {
+        'content-type': 'application/json'
+      },
+      body: JSON.stringify({
+        account_id: accountId
+      })
+    })
+    const responseText = await response.text()
+    const responseBody = JSON.parse(responseText)
+    return responseBody
```
Comment (swansontec) [inline src/filecoin/Filscan.ts:95]: We aren't cleaning anything in the Filscan subsystem. What would be the consequences if they return something else, like `[ "LOL, we shut the server down" ]`? We generally like to see cleaners whenever data comes in from the wild wild internet.

**Review by swansontec — APPROVED** [review]
I mostly have questions. Maybe some of them need to get fixed.

File: src/filecoin/FilecoinTools.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    if (walletInfo.type !== this.currencyInfo.walletType) {
+      throw new Error('InvalidWalletType')
+    }
+
+    const { pluginId } = this.currencyInfo
+    const { hdPathCoinType } = this.networkInfo
+
+    const filecoinPrivateKeys = asFilecoinPrivateKeys(pluginId)(walletInfo.keys)
+
+    // TODO: Figure out how to use the accountData.privateKey Buffer to gen xpub
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/filecoin/FilecoinTools.ts:101]: No it hasn't. The filecoin library doesn't have any public xpub fields for the account data it returns. So I just derive the address using the bip32 lib manually. It'd be cleaner to derive once using the filecoin lib; perhaps we can fork the lib and include xpub data in the `AccountData` object.

**Review by swansontec — APPROVED** [review]
Two small things.

**Pull request merged**